### ISSUE TITLE
Forms not working properly

### ISSUE THREAD

**Issue by j245** [statamic/v2-hub#677]
### Steps to reproduce
1. Setup a form with a `{{ if {form:success} }}` conditional (and a `redirect="{current_url}/thanks"`. Set up your `routes.yaml` to `ignore: - thanks` - although it still occurs regardless of whether or not we use the redirect)
2. Submit a form
3. Sit back and gasp as the following behaviour occurs...

### Expected behaviour
The form should submit, send an email and take you to `/thanks` and show the success message.

### Actual behaviour
You're taken to `/thanks` no success message appears, no email is sent. Also sometimes if you load the form page later on, it shows you the success message, even though you submitted that form ages ago and have since browsed elsewhere.

### Where it happens
This only happens on our production server, not our staging (similar to production with load balancer) or development (MAMP) setup. Thought it was a `2.0.4` thing but happens on `2.0.3` too.

### Operating system:
Ubuntu 14.04.4 LTS
### Web server:
Apache 2.4.7
### PHP version:
5.5.9
### Statamic version:
2.0.3 and 2.0.4
### Updated from an older Statamic or fresh install:
Updated (but was occurring on 2.0.0) as well
### List of installed addons:
Bespoke google re-captcha add-on

#### Web server error log
```
Couldn't see any.
```

#### Statamic error log (`/local/storage/logs/statamic.log`)
```
Will try and get hold of these at some point
```

**Comment by j245** [comment]
We think this is actually to do with the production environment setup. We have a load balancer in front of the server running Statamic. Behind the load balance the site is duplicated on two servers.

We looked at the access logs and found that if someone was hitting the same server for the whole journey then it was succeeding. It failed when the user got moved to the other server half way through the journey.

In our QA environment we also have a load balancer but only 1 server behind it. This setup never goes wrong.
